blog/en-us/introduction-to-dubbo-qos.md

```diff
@@ -0,0 +1,248 @@
+# Manipulating Services Dynamically via QoS
+
+QoS (short form of **Quality of Service**), is a common terminology talking about network devices. For example, by adjusting and manipulating the weights of ports of a router dynamically via QoS, engineers could give priority to services running on these ports and make sure these services' quality and reliability.
+
+In Dubbo, QoS is used to query and manipulate services dynamically, like getting a list of active provider and consumer services, and launching or withdrawing services (i.e registering to or unregistering services from registration center).
+
+## Mechanism of QoS
+
+From 2.5.8, QoS is introduced into Dubbo and is activated by default. All QoS's features are abstracted to commands, which could be executed to get responses from QoS.
+
+> QoS is based on Netty4. In versions earlier than 2.6.x, Dubbo relies on Netty3, so you have to add Netty4 as a dependency explicitly to ensure that Netty4 works. If you generate a Dubbo application on http://start.dubbo.io, there's no need to add configurations because Netty4 is listed as a dependency by default.
+
+![undefined](../../img/blog/qos-architecture.png)
+
+The picture above shows how QoS works:
+
+1. Start and listen to a port (22222 by default).
+2. Choose a corresponding request handler by detecting the protocol (telnet or http) a request comply with.
+3. Decode and parse the request to generate corresponding command according to the protocol.
+4. Execute commands and return with responses.
+
+### QoS Commands:
+
+Commands that QoS supports at the current moment include:
+
+* `help`, list available commands
+* `ls`: list all active provider services and consumer services
+* `online`: dynamically register a service or all services to registration center
+* `offline`: dynamically remove (unregister) a services or all services from registration center
+* `quit`: quit the current telnet session
+
+Now we are going to demonstrate how to manipulate services dynamically via QoS.
+
+#### Access QoS via Telnet
+
+Assuming that our Dubbo server has started, connect to it via telnet:
+
+```shell
+$ telnet localhost 22222
+Trying 127.0.0.1...
+Connected to localhost.
+Escape character is '^]'.
+  ?????????  ???    ??  ???????????  ???????????   ????????
+  ???   ???? ???    ???   ???    ???   ???    ??? ???    ???
+  ???    ??? ???    ???   ???    ???   ???    ??? ???    ???
+  ???    ??? ???    ???  ??????????   ??????????  ???    ???
+  ???    ??? ???    ??? ???????????  ???????????  ???    ???
+  ???    ??? ???    ???   ???    ???   ???    ??? ???    ???
+  ???   ???? ???    ???   ???    ???   ???    ??? ???    ???
+  ?????????  ?????????  ???????????  ???????????   ????????
```

Comment:
Why so many ???

Reply:
You can check [the Chinese version](https://github.com/apache/incubator-dubbo-website/blob/asf-site/blog/zh-cn/introduction-to-dubbo-qos.md). It's just the command displaying a 'DUBBO'.